flik/trustregion/cauchy.py

```diff
@@ -0,0 +1,86 @@
+"""Cauchy-point method for the subproblem of the trust-region method."""
+
+import numpy as np
+
+from subspace import SubProblem
+
+
+class Cauchy(SubProblem):
+    """The Cauchy-point method.
+
+    Attributes
+    ----------
+    point: np.ndarray (1-dimensional)
+        Initial point of the optimization.
+    func: callable
+        Objective function.
+    grad: callable
+        Gradient of the objective function.
+    hessian: callable
+        Hessian of the objective function.
+    radius: float
+        The trust region radius.
+
+    Methods
+    -------
+    __init__(self, point, func, grad, hess, radius, *params)
+        Initialize the data corresponding to the given point.
+    get_scaling(self)
+        Compute the scaling factor.
+    solver(self)
+        Subproblem Dogleg solver.
+    """
+
+    def __init__(self, point, func, grad, hess, radius):
+        """Initialize the method.
+
+        Parameters
+        ----------
+        point: np.ndarray((N,))
+            Initial point of the optimization.
+        func: callable
+            Objective function.
+        grad: callable
+            Gradient of the objective function.
+        hess: callable
+            Hessian of the objective function.
+        radius: float
+            The trust region radius.
+        """
+        super().__init__(point, func, grad, hess, radius)
+        self.gradient = self.grad(self.point)
+        self.hessian = self.hess(self.point)
+
+    def get_scaling(self):
+        r"""Define the scaling-factor of the step.
+
+        :math:`\tao = 1` if :math:`g^T B g \leq 0`
```

Comment:
I believe the Greek letter is actually called tau instead of tao.

Reply:
ohh yes, you are right, thanks!